src/pages/template.tsx

```diff
@@ -0,0 +1,230 @@
+"use client";
+
+import { Button } from "@/components/ui/button";
+import { ArrowDown, Mail, AlertCircle, HelpCircle } from "lucide-react";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import * as React from "react";
+
+export function ContactButton() {
+  return (
+    <div className="button-container">
+      <Button variant="contactButton1" size="contactButtonSize">
+        <ArrowDown size={24} />
+        GET IN TOUCH
+        <ArrowDown size={24} />
+      </Button>
+      <Button variant="contactButton2" size="contactButtonSize">
+        <ArrowDown size={24} />
+        GET IN TOUCH
+        <ArrowDown size={24} />
+      </Button>
+      <Button variant="contactButton3" size="contactButtonSize">
+        <ArrowDown size={24} />
+        GET IN TOUCH
+        <ArrowDown size={24} />
+      </Button>
+      <Button variant="contactButton4" size="contactButtonSize">
+        <ArrowDown size={24} />
+        GET IN TOUCH
+        <ArrowDown size={24} />
+      </Button>
+    </div>
+  );
+}
+
+export function InputWithLabel() {
+  const [value, setValue] = React.useState(""); // Tracks the input value
+  const [isFocused, setIsFocused] = React.useState(false); // Tracks the focus state
+  const [error, setError] = React.useState(false); // Tracks the error state
+  // const [showHint, setShowHint] = React.useState(false); // Tracks hint visibility
+
+  // Validation: Valid if the input contains "@"
+  const validateInput = (inputValue: string) => {
+    return String(inputValue)
+      .toLowerCase()
+      .match(
+        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+      );
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = e.target.value;
+    setValue(inputValue);
+
+    // Validate input on change
+    setError(inputValue !== "" && !validateInput(inputValue));
+  };
+
+  const handleBlur = () => {
+    setIsFocused(false);
+
+    // Validate input on blur
+    if (value !== "") {
+      setError(!validateInput(value));
+    }
+  };
+
+  const handleFocus = () => {
+    setIsFocused(true);
+  };
+
+  return (
+    <div className="space-y-6">
+      {/* Input Container */}
+      <div className="relative max-w-[320px] font-Montserrat">
+        <Label htmlFor="email" className="text-sm font-medium leading-[20px] text-[#0A3D62]">
```

Comment:
@rakeem996 This whole block needs to be a component so that every other page simple has to add this and they will get a fully functionnig input with the correct theme, error state etc in their page without having to repeat this code every single time.

```html
<Input type="email" id="email" placeholder="Email" />
```